feat: implement onboarding (#928)
DisnakeDev/disnake: docs/api.rst

        See :attr:`automod_block_message` for more information on how the
        :attr:`~AuditLogEntry.extra` field is set.

    .. attribute:: onboarding_prompt_create

        An onboarding prompt was created.

        When this is the action, the type of :attr:`~AuditLogEntry.target` is
        the :class:`Object` with the ID of the onboarding prompt which
        was created.

>         Possible attributes for :class:`AuditLogDiff`:

Reviewer: This seems to include the `type` field as well

        When this is the action, the type of :attr:`~AuditLogEntry.target` is
        the :class:`Object` with the ID of the onboarding prompt which
        was updated.

        Possible attributes for :class:`AuditLogDiff`:

        - :attr:`~AuditLogDiff.title`
        - :attr:`~AuditLogDiff.options`
        - :attr:`~AuditLogDiff.single_select`
        - :attr:`~AuditLogDiff.required`
        - :attr:`~AuditLogDiff.in_onboarding`

    .. attribute:: onboarding_prompt_delete

        An onboarding prompt was deleted.

        When this is the action, the type of :attr:`~AuditLogEntry.target` is
        the :class:`Object` with the ID of the onboarding prompt which
        was deleted.

        Possible attributes for :class:`AuditLogDiff`:

        TBD.

    .. attribute:: onboarding_create

        An onboarding was created.

        When this is the action, the type of :attr:`~AuditLogEntry.target` is
        ``None``.

        Possible attributes for :class:`AuditLogDiff`:

        - :attr:`~AuditLogDiff.default_channels`
        - :attr:`~AuditLogDiff.enable_default_channels`
        - :attr:`~AuditLogDiff.enable_onboarding_prompts`

>         - :attr:`~AuditLogDiff.prompts`

Reviewer: ```suggestion
        - :attr:`~AuditLogDiff.prompts`
        - :attr:`~AuditLogDiff.enabled`
```

    .. attribute:: onboarding_update

        An onboarding was updated.

        When this is the action, the type of :attr:`~AuditLogEntry.target` is
        ``None``.

        Possible attributes for :class:`AuditLogDiff`:

>         TBD.

Reviewer: This appears to match the list for `onboarding_create`

        :type: Optional[:class:`ThreadSortOrder`]

    .. attribute:: title

        The title of an onboarding prompt or an onboarding prompt option being changed.

        :type: :class:`str`

    .. attribute:: options

        The list of options of an onboarding prompt being changed.

        :type: List[:class:`OnboardingPromptOption`]

    .. attribute:: single_select

        The onboarding prompt is single select or not.

        :type: :class:`bool`

    .. attribute:: required

        The onboarding prompt option is required or not.

        :type: :class:`bool`

    .. attribute:: in_onboarding

>         The onboarding prompt option is in onboarding or not.

Reviewer: ```suggestion
        Whether the onboarding prompt option is part of the initial onboarding flow.
```

        :type: :class:`bool`

    .. attribute:: default_channel

        The list of default channels of an onboarding being changed.

>         :type: List[:class:`abc.GuildChannel`]

Reviewer: Duplicate of `default_channels` below

    .. attribute:: enable_default_channels

        The onboarding shows default channels or not.

        :type: :class:`bool`

    .. attribute:: enable_onboarding_prompts

        The onboarding shows onboarding prompts or not.

>         :type: :class:`bool`

Reviewer: These aren't documented

>     .. attribute:: prompts
> 
>         The list of prompts of an onboarding being changed.
> 
>         :type: List[:class:`OnboardingPrompt`]
> 
>     .. attribute:: default_channels
> 
>         The list of default channels of an onboarding being changed.
> 
>         :type: List[:class:`~.abc.GuildChannel`]

Reviewer: ```suggestion
    .. attribute:: prompts

        The list of onboarding prompts being changed.

        :type: List[:class:`OnboardingPrompt`]

    .. attribute:: default_channels

        The list of default channels for onboarding being changed.

        :type: List[:class:`~.abc.GuildChannel`]
```

Author: closing since audit logs were not added in api-docs pr. 2b81f4c90af51506ae7f6c0843f514b58be80f7a

.. attributetable:: ApplicationRoleConnectionMetadata

> .. autoclass:: ApplicationRoleConnectionMetadata

Reviewer: ```suggestion
.. autoclass:: ApplicationRoleConnectionMetadata
    :members:
```
looks like this one disappeared when the other two got added

.. autoclass:: ApplicationRoleConnectionMetadata
    :members:

OnboardingPrompt
~~~~~~~~~~~~~~~~

.. attributetable:: OnboardingPrompt

.. autoclass:: OnboardingPrompt
    :members:

OnboardingPromptOption
~~~~~~~~~~~~~~~~~~~~~~

.. attributetable:: OnboardingPromptOption

.. autoclass:: OnboardingPromptOption
    :members:

> 

Reviewer: these two should be put under Discord Models, I think

Author: c2ce2cc76acec06a265cfdc38e811315aeceb856
